pkg/server/server.go

```diff
@@ -279,6 +287,7 @@ func stageFiles(ctx context.Context, sc *Context, controlConfig *config.Control)
 		"%{SYSTEM_DEFAULT_REGISTRY}%":     registryTemplate(controlConfig.SystemDefaultRegistry),
 		"%{SYSTEM_DEFAULT_REGISTRY_RAW}%": controlConfig.SystemDefaultRegistry,
 		"%{PREFERRED_ADDRESS_TYPES}%":     addrTypesPrioTemplate(controlConfig.FlannelExternalIP),
+		"%{DATA_DIR}%":                    CNIBinDir,
```

Comment:
Lets not call this `DATA_DIR` - this could be confusing as normally that would refer to the value of the `--data-dir` flag. 
```suggestion
		"%{CNI_BIN_DIR}%":                    CNIBinDir,
```

Reply:
There are other problems with injecting the server's CNI bin dir path into the chart like this though:
1. CNI bin dir is version dependent. The value will change whenever the server is upgraded, and will only work if all nodes are on the exact same build of k3s:
  ```console
  brandond@seago:~$ grep bin_dir /var/lib/rancher/k3s/agent/etc/containerd/config.toml
  bin_dir = "/var/lib/rancher/k3s/data/0bd989dbe82165818ee98da4cc70b043adabccc23827c9bbd24b1460d4f3f2e1/bin"
  ```
2. CNI bin dir is prefixed by --data-dir, which may be set to different values on different hosts.

You're going to need to find some way to set this on per-node basis, instead of injecting it via the chart. You could MAYBE get away with hardcoding it as `/var/lib/rancher/k3s/data/current` but this only fixes problem 1, not problem 2.

Reply:
Thanks Brad, those are great points I did not think about